### PR TITLE
Expose connection with Documentation parser and generator in ScalaJS artifact

### LINKED ISSUE #1001: Compile Documentation Parser to ScalaJS
### Summary
As currently doc parser is able to parse and generate code, we need to be able to use it in IDE, so it needs to be compiled to scalaJS, like the Enso Parser.

### Value
Doc Parser is useable

### Specification
Doc parser provides a function which parse a documentation string and returns a HTML-formatted string. The function should be exposed by scala-parser.js artifact.

### Acceptance Criteria & Test Cases
Doc Parser works as expected - it is integrated with IDE and there are tests in IDE showing that the integration works.

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
Closes #1001
Enables working with Doc Parser from IDE with rust.
IDE runs `doc_parser_generate_html_source` with source code of selected node, and it gives back HTML file of formatted documentation to present in IDE's documentation controller.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: lib/scala/syntax/specialization/js/src/main/scala/org/enso/syntax/text/Main.scala [hunk 1]
```diff
@@ -31,4 +35,19 @@ object Parse {
...
+  def doc_parser_generate_html_source(program: String): String = {
+    try {
+      val parser   = new Parser()
+      val module   = parser.run(program)
+      val dropMeta = parser.dropMacroMeta(module)
+      val doc      = DocParserRunner.createDocs(dropMeta)
+      val htmlCode = DocParserHTMLGenerator.generateHTMLForEveryDocumented(doc)
+      htmlCode
+    } catch {
+      // FIXME We wrap the error message in JavaScriptException, so that javascript
```
Comment (iamrecursion) [inline lib/scala/syntax/specialization/js/src/main/scala/org/enso/syntax/text/Main.scala]: We're on ScalaJS 1.x, so please test if this is still needed.

**Review by iamrecursion — CHANGES_REQUESTED** [review]

File: lib/scala/syntax/specialization/js/src/main/scala/org/enso/syntax/text/Main.scala (hunk 1 above)
Comment (BinarySoftware) [reply lib/scala/syntax/specialization/js/src/main/scala/org/enso/syntax/text/Main.scala]: Tested with IDE, seems to work just fine

Comment (BinarySoftware) [reply lib/scala/syntax/specialization/js/src/main/scala/org/enso/syntax/text/Main.scala]: And tested in browser, produces HTML code from documented code, just as expected.

**Review by iamrecursion — APPROVED** [review]
Good stuff!

**Pull request merged**